Automated Licensing Documentation

quazirafi opened the pull request:
1. Added instructions for developers to add license headers automatically to the source programs file of newly created repositories. 
2. Also, added the list of repos that follow the 4-layered CI Architecture.

twindus reviewed (commented):
Overall, looks good, just needs a couple of simple mods.

Need to spell workflow correctly (currently at least one instance of wrokflow).

Change "Everything will remain same" to "Everything will remain the same"

hjjvandam commented:
Overall this looks good. I have a few minor points though (going sequentially through the text from beginning to end):
- Instead of "all our source codes" I think "all our source code" is better.
- I guess the path ".github/.github/workflows/add_licenses_master.yaml" contains a ".github" too many.
- Instead of "You may recall that it is" I think "You may recall that this is" is more natural English.
- Instead of "You do not need to do any change" I would write "You do not need to make any changes".
- I would write "file from the ``NWChemEx`` repository" instead of "file from ``NWChemEx`` repository".
- The "wrokflow" was mentioned by Theresa already.
- Again, I would add "the" before NWChemEx in "``.licenserc.yaml`` file from ``NWChemEx``".
- Instead of "may cause error." I would write "may cause errors."
- I would replace "all source codes in the repo" with "all source files in the repo" which is a little bit clearer in this context, I think.

Otherwise this is very nice.

quazirafi commented:
> Overall, looks good, just needs a couple of simple mods.
> 
> Need to spell workflow correctly (currently at least one instance of wrokflow).
> 
> Change "Everything will remain same" to "Everything will remain the same"

@hjjvandam will change those! Thanks!

ryanmrichard commented on docs/source/project_management/ci_maintainer_notes.rst:
```diff
@@ -220,3 +234,21 @@ The ``test_docs.yaml`` file will call the ``deploy_docs_tmpl.yaml`` file of the
+
+NWChemEx uses `Apache License (2.0) <https://www.apache.org/licenses/LICENSE-2.0>`__. We want all our source codes (except for some configuration, test and input files) to have the proper license header when deployed. We have workflow files for automatically adding the license header. The main workflow file resides in ``.github/.github/workflows/add_licenses_master.yaml`` which is in the ``.githu ...
+
+1. Firstly, you can copy the ``add_licenses.yaml`` file from any other repo ( except ``.github`` ) listed in the very beginning of this documentation and paste it in your new repository's ``.github/workflows`` path which will be something like ``new-repo-name/.github/workflows/``. For example, you can copy the ``add_licenses.yaml`` file from ``NWChemEx`` repository which resides in the path ``NWC ...
```
```suggestion
1. Firstly, you can copy the ``add_licenses.yaml`` file from any other repo ( except ``.github`` ) listed in the very beginning of this documentation and paste it in your new repository's ``.github/workflows`` path which will be something like ``new-repo-name/.github/workflows/``. For example, you can copy the ``add_licenses.yaml`` file from ``NWChemEx`` repository which resides in the path ``NWChemEx/.github/workflows/add_licenses.yaml`` of the repository. This workflow file calls the ``add_licenses_master.yaml`` workflow file which resides in the ``.github`` repo with two parameters: ``config_file`` and ``token``. The value for ``token`` should be ``{{ secrets.GITHUB_TOKEN }}``, which is same for every repository.
```

ryanmrichard replied:
@hjjvandam and @twindus if you select a line (or lines) of code and then click the page icon with the $\pm$ icon you can suggest the changes yourself.

hjjvandam replied:
OK, good to know.

twindus replied:
I guess I knew that, but forgot!  :)

quazirafi commented:
> Overall, looks good, just needs a couple of simple mods.
> 
> Need to spell workflow correctly (currently at least one instance of wrokflow).
> 
> Change "Everything will remain same" to "Everything will remain the same"

Sorry for the spelling! Will fix that! Thanks!

quazirafi commented:
> Overall this looks good. I have a few minor points though (going sequentially through the text from beginning to end):
> 
> * Instead of "all our source codes" I think "all our source code" is better.
> * I guess the path ".github/.github/workflows/add_licenses_master.yaml" contains a ".github" too many.
> * Instead of "You may recall that it is" I think "You may recall that this is" is more natural English.
> * Instead of "You do not need to do any change" I would write "You do not need to make any changes".
> * I would write "file from the `NWChemEx` repository" instead of "file from `NWChemEx` repository".
> * The "wrokflow" was mentioned by Theresa already.
> * Again, I would add "the" before NWChemEx in "`.licenserc.yaml` file from `NWChemEx`".
> * Instead of "may cause error." I would write "may cause errors."
> * I would replace "all source codes in the repo" with "all source files in the repo" which is a little bit clearer in this context, I think.
> 
> Otherwise this is very nice.

@hjjvandam 
1. Fixed
2. Actually, it is correct I suppose. .github is the name of the repository. Then we have the workflow files inside of the .github/workflows directory inside of the .github repository. 
![Screen Shot 2022-09-16 at 3 45 52 PM](https://user-images.githubusercontent.com/15170739/190730020-525caf0b-d599-4b62-b04c-2a4087d0e8de.png)
3. Fixed
4. Fixed
5. Fixed
6. Fixed
7. Fixed
8. Fixed
9. Fixed

quazirafi commented:
> Overall, looks good, just needs a couple of simple mods.
> 
> Need to spell workflow correctly (currently at least one instance of wrokflow).
> 
> Change "Everything will remain same" to "Everything will remain the same"

@twindus fixed!
